Importer/EventListener/ErrorSubscriber.php

```diff
@@ -0,0 +1,191 @@
+use Netdudes\ImporterBundle\Importer\Error\CsvHeadersError;
+use Netdudes\ImporterBundle\Importer\Error\Error;
+use Netdudes\ImporterBundle\Importer\Error\FileNotFoundError;
+use Netdudes\ImporterBundle\Importer\Event\ApplicationErrorEvent;
+use Netdudes\ImporterBundle\Importer\Event\ConfigurationErrorEvent;
+use Netdudes\ImporterBundle\Importer\Event\DataErrorEvent;
+use Netdudes\ImporterBundle\Importer\Event\ImportEvents;
+use Netdudes\ImporterBundle\Importer\Interpreter\Exception\DateTimeFormatException;
+use Netdudes\ImporterBundle\Importer\Interpreter\Exception\InvalidEntityException;
+use Netdudes\ImporterBundle\Importer\Interpreter\Exception\LookupFieldException;
+use Netdudes\ImporterBundle\Importer\Interpreter\Exception\RowSizeMismatchException;
+use Netdudes\ImporterBundle\Importer\Interpreter\Exception\UnknownColumnException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Validator\ConstraintViolation;
+
+class ErrorSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ImportEvents::APPLICATION_ERROR => ['handleApplicationError'],
+            ImportEvents::CONFIGURATION_ERROR => ['handleConfigurationError'],
+            ImportEvents::DATA_ERROR => ['handleDataError'],
+        ];
+    }
+
+    /**
+     * @param ApplicationErrorEvent $event
+     */
+    public function handleApplicationError(ApplicationErrorEvent $event)
+    {
+        $exception = $event->getException();
+        $log = $event->getLog();
+
+        $message = sprintf(
+            '%s: %s (uncaught exception) at %s line %d while running import.',
+            get_class($exception),
+            $exception->getMessage(),
+            $exception->getFile(),
+            $exception->getLine()
+        );
+
+        $log->addCriticalError($message);
+    }
+
+    /**
+     * @param ConfigurationErrorEvent $event
+     */
+    public function handleConfigurationError(ConfigurationErrorEvent $event)
+    {
+        $error = $event->getError();
+        $log = $event->getLog();
+
+        switch (true) {
+            case ($error instanceof CsvHeadersError):
+                $message = $error->getMessage();
+                break;
+            case ($error instanceof FileNotFoundError):
+                $message = $error->getMessage();
+                break;
+            case ($error instanceof Error):
+                $message = $this->buildErrorMessage($error);
+                break;
+            default:
+                $message = 'An unknown error occurred when uploading the file.';
+        }
+
+        $log->addConfigurationError($message);
+    }
+
+    /**
+     * @param DataErrorEvent $event
+     */
+    public function handleDataError(DataErrorEvent $event)
+    {
+        $log = $event->getLog();
+        $exception = $event->getException();
+
+        switch (true) {
```

Comment:
Technically, we could:

```
try {
    throw $exception;
} catch (DateTimeFormatException $exception) {
    // handle me
}
// and all other types
}
```

But probably won't improve readability, so might not be worth it.